sympy/core/expr.py

```diff
@@ -1059,7 +1059,7 @@ def dir(self, x, cdir):
 
     def _eval_transpose(self):
         from sympy.functions.elementary.complexes import conjugate
-        if (self.is_complex or self.is_infinite):
+        if (self.is_complex or self.is_infinite or self.is_commutative):
```

Comment:
We can remove `is_complex` and `is_infinite` here because `complex` implies `commutative` and `is_infinite` is just trying to allow things like `oo` for which `is_complex` would be False.